src/compiler/checker.ts

```diff
@@ -4057,12 +4111,27 @@ module ts {
                     if (source === nullType && target !== undefinedType) return Ternary.True;
                     if (source.flags & TypeFlags.Enum && target === numberType) return Ternary.True;
                     if (source.flags & TypeFlags.StringLiteral && target === stringType) return Ternary.True;
+                    if (source.flags & TypeFlags.TypeGuard && target.flags & TypeFlags.Boolean) return Ternary.True;
                     if (relation === assignableRelation) {
                         if (source.flags & TypeFlags.Any) return Ternary.True;
                         if (source === numberType && target.flags & TypeFlags.Enum) return Ternary.True;
                     }
                 }
                 let saveErrorInfo = errorInfo;
+                if (source.flags & TypeFlags.TypeGuard && target.flags & TypeFlags.TypeGuard) {
+                    if ((<TypeGuardType>source).parameterIndex &&
+                        (<TypeGuardType>target).parameterIndex &&

```

Comment:
If either parameterIndex is 0, this will pass. I think you mean `parameterIndex >= 0`